server/instrumentation-backend/src/sh/calaba/instrumentationbackend/actions/text/DeleteSurroundingText.java

```diff
@@ -0,0 +1,86 @@
+package sh.calaba.instrumentationbackend.actions.text;
+
+import android.os.Build;
+import android.text.Editable;
+import android.text.Selection;
+import android.view.inputmethod.BaseInputConnection;
+import android.view.inputmethod.InputConnection;
+
+import sh.calaba.instrumentationbackend.InstrumentationBackend;
+import sh.calaba.instrumentationbackend.Result;
+import sh.calaba.instrumentationbackend.actions.Action;
+
+public class DeleteSurroundingText implements Action {
+
+    private static final String USAGE = "This action takes 2 arguments:\n([int] beforeLength, [int] afterLength)";
+
+    @Override
+    public Result execute(final String... args) {
+        if (args.length != 2) {
+            return Result.failedResult(USAGE);
+        } 
+
+        final InputConnection inputConnection = InfoMethodUtil.tryGetInputConnection();
+
+        if (inputConnection == null) {
+            return Result.failedResult("Unable to set selection, no element has focus");
+        }
+
+        if (!(inputConnection instanceof BaseInputConnection)) {
```

Comment:
I thought `BaseInputConnection` was the base class of all input connections - would `GetInputConnection()` ever return something else?